identityClient/api.go

```diff
@@ -226,6 +226,26 @@ type InternalIdentityLoginResponse struct {
 	UserID    string `json:"user_id"`    // The ID of the Identity's Keycloak user.
 }
 
+// InternalIdentityStatusResponse wraps an internal Identity
+// status context for realm identities.
+type InternalIdentityStatusResponse struct {
+	Locked bool `json:"locked"` // Whether the Identity's account is currently locked for logging into a realm.
+}
+
+// InternalIdentityStatusUserIdRequest wraps the parameters
+// needed to determine the status of an Identity from its Identity User ID.
+type InternalIdentityStatusUserIdRequest struct {
+	RealmName string    // The name of the realm the Identity is a member of.
```

Comment:
Reminder to use RealmDomainName everywhere as thats the internal service identifier (vs the realm name which could be upper or mixed cased)

Reply:
Ah yes, I need to make sure I clarify that for myself. I will go through all my PRs and update this naming

Reply:
@galxy25 Should we use RealmDomain to keep consistent with other structs